---
version: "3.9"

services:
  caddy:
    build:
      context: .
      dockerfile: Dockerfile
    image: caddy-docker-proxy:v2.6.2
    container_name: caddy-docker-proxy
    restart: unless-stopped
    labels:
      caddy_0.acme_dns: "cloudflare {env.CF_API_TOKEN}"
      caddy_0.email: "{env.EMAIL}"

      caddy_1: (fa_authentik)
      caddy_1.forward_auth: https://auth.{env.DOMAIN}
      caddy_1.forward_auth.uri: /outpost.goauthentik.io/auth/caddy
      caddy_1.forward_auth.copy_headers: X-Authentik-Username X-Authentik-Groups X-Authentik-Email X-Authentik-Name X-Authentik-Uid X-Authentik-Jwt X-Authentik-Meta-Jwks X-Authentik-Meta-Outpost X-Authentik-Meta-Provider X-Authentik-Meta-App X-Authentik-Meta-Version
      caddy_1.forward_auth.header_up: Host {upstream_hostport}
    env_file:
      - .env
    environment:
      - CADDY_INGRESS_NETWORKS=cdp
    volumes:
      # for caddy-docker-proxy to work
      - /var/run/docker.sock:/var/run/docker.sock
      # for caddy data and config files
      - ~/selfhosted/data/caddy-docker-proxy/data:/data
      - ~/selfhosted/data/caddy-docker-proxy/config:/config
    ports:
      - "4567:443"
      - "4567:443/udp"
    networks:
      - cdp

networks:
  cdp:
    external: true